src/Microsoft.VisualStudio.Threading/AsyncLazy.cs

```diff
@@ -147,19 +147,55 @@ public Task<T> GetValueAsync(CancellationToken cancellationToken)
                         resumableAwaiter = new InlineResumable();
                         Func<Task<T>> originalValueFactory = this.valueFactory;
                         this.valueFactory = null;
-                        Func<Task<T>> valueFactory = async delegate
+
+                        Task<T> ValueFactory()
```

Comment:
Do we have to change the syntax from an anonymous delegate to a nested method here? I prefer the old syntax when we actually need a delegate (which we do) rather than a method, and also we're defining this in the middle of a block. I like the nested method syntax when they are helper-like methods that can be defined at the bottom of the method.
Given this is for 15.8, evidently, I think we should minimize risk too. I personally am not 100% confident that we haven't changed the closure semantics by switching to nested method syntax.